lib/espresso-runner.js

```diff
@@ -41,6 +41,12 @@ class EspressoRunner {
   }
 
   async installServer () {
+    // this.adb.installOrUpgrade remains newer Espresso server
+    const appState = await this.adb.getApplicationInstallState(this.modServerPath, TEST_APK_PKG);
+    if (appState === this.adb.APP_INSTALL_STATE.NEWER_VERSION_INSTALLED) {
+      await this.adb.uninstallApk(TEST_APK_PKG);
+    }
+
     await this.adb.installOrUpgrade(this.modServerPath, TEST_APK_PKG);
```

Comment:
regarding the cost of getApplicationInstallState call - it is called inside installOrUpgrade for the second time. This means we should consider changing `installOrUpgrade` to `install`, like UIA2 driver does if the upgrade logic needs to be customized

Reply:
yeah..